daemon/daemon.go

```diff
@@ -785,10 +789,24 @@ func (daemon *Daemon) initDiscovery(config *Config) error {
 // - Daemon max concurrent uploads
 // - Cluster discovery (reconfigure and restart).
 func (daemon *Daemon) Reload(config *Config) error {
+	var err error
+	// used to hold reloaded changes
+	attributes := map[string]string{}
+
+	// We need defer here to ensure the lock is released as
```

Comment:
If we put this defer after `defer daemon.configStore.reloadLock.Unlock()` the lock would still be held when calling `daemon.LogDaemonEventWithAttributes()` which would cause a deadlock